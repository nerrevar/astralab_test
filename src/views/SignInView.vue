<template>
  <div class="wrapper">
    <div class="title">
      Sign In
    </div>
    <FormTemplate
      :formFields="formFields"
      submitText="Sign In"
      @submit="submit($event)"
    />
    <div class="sign-up">
      Donâ€™t have an account yet?
      <br />
      <router-link to="/sign_up">
        Sign Up
      </router-link>
    </div>
    <div
      class="login-error"
      v-show="loginErrorVisible"
    >
      <span class="login-error__text">
        Wrong email or password
      </span>
      <img
        class="login-error__close"
        src="@/assets/close.svg"
        alt="X"
        @click="setLoginErrorVisible(false)"
      />
    </div>
  </div>
</template>

<script>
import { mapMutations, mapGetters } from 'vuex'

import FormTemplate from '@/components/FormTemplate.vue'

import {
  emailFormFieldParams,
  passwordFormFieldParams
} from '@/utils'

export default {
  name: 'SignInView',
  components: { FormTemplate },
  data () {
    return {
      formFields: [
        emailFormFieldParams,
        passwordFormFieldParams
      ],
      loginErrorVisible: false,
    }
  },
  computed: mapGetters(['getFirebaseApp']),
  methods: {
    ...mapMutations(['setUser']),
    setLoginErrorVisible (val) {
      this.loginErrorVisible = val
    },
    submit (data) {
      this.getFirebaseApp.auth().signInWithEmailAndPassword(data.email, data.password)
        .then(userCredential => {
          const { displayName, email } = userCredential.user
          this.setUser({
            user: {
              displayName,
              email,
            },
          })
          this.$router.push('/main')
        })
        .catch(() => {
          this.loginErrorVisible = true
          setTimeout(
            () => {
              this.loginErrorVisible = false
            },
            10000
          )
        })
    },
  },
}
</script>

<style lang="sass" scoped>
.sign-up
  text-align: center
  margin-top: 2em

  & > a
    color: black

.login-error
  +ellipse
  position: absolute
  right: 12px
  bottom: 12px
  left: 12px
  height: 48px
  display: flex
  flex-flow: row nowrap
  justify-content: space-between
  background-color: #ff6683
</style>
