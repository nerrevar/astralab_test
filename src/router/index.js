import Vue from 'vue'
import VueRouter from 'vue-router'

import EnterView from '@/views/EnterView.vue'
import SignUpView from '@/views/SignUpView.vue'
import SignInView from '@/views/SignInView.vue'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'EnterView',
    component: EnterView,
  },
  {
    path: '/sign_up',
    name: 'SignUpView',
    component: SignUpView,
  },
  {
    path: '/sign_in',
    name: 'SignInView',
    component: SignInView,
  },
  {
    path: '*',
    redirect: '/',
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
})

export default router
