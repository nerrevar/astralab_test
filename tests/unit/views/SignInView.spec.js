import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'

import router from '@/router'

import SignInView from '@/views/SignInView.vue'

const localVue = createLocalVue()

localVue.use(VueRouter)

describe('SignInView.vue', () => {
  const vm = shallowMount(
    SignInView,
    {
      localVue,
      router,
    }
  )

  it('renders', () => {
    expect(vm.element).toMatchSnapshot()
  })

  it('setLoginErrorVisible true', () => {
    expect(vm.find('.login-error').isVisible()).toBeFalsy()
    vm.vm.setLoginErrorVisible(true)
    vm.vm.$nextTick(() => {
      expect(vm.find('.login-error').isVisible()).toBeTruthy()
    })
  })

  it('method submit', () => {
    expect(vm.vm.submit()).toBeUndefined()
  })
})
