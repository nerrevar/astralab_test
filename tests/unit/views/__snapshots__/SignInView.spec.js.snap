// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInView.vue renders 1`] = `
<div
  class="wrapper"
>
  <div
    class="title"
  >
    
    Sign In
  
  </div>
   
  <formtemplate-stub
    formfields="[object Object],[object Object]"
    submittext="Sign In"
  />
   
  <div
    class="sign-up"
  >
    
    Donâ€™t have an account yet?
    
    <br />
     
    <router-link-stub
      ariacurrentvalue="page"
      event="click"
      tag="a"
      to="/sign_up"
    >
      
      Sign Up
    
    </router-link-stub>
  </div>
   
  <div
    class="login-error"
    style="display: none;"
  >
    <span
      class="login-error__text"
    >
      
      Wrong email or password
    
    </span>
     
    <img
      alt="X"
      class="login-error__close"
      src="@/assets/close.svg"
    />
  </div>
</div>
`;
